tests/core/component.test.js

```diff
@@ -266,4 +266,40 @@ suite('Component', function () {
       assert.equal(component.data.color, 'blue');
     });
   });
+
+  suite('update', function () {
+    setup(function () {
+      components.dummy = undefined;
+    });
+
+    test('not called if component data does not change', function () {
+      var updateStub = sinon.stub();
+      var TestComponent = registerComponent('dummy', {
+        schema: {color: { default: 'red' }},
```

Comment:
I don't have much issues mixing up blocks and object literals. We already use spaces for literals and If I change it here I have to change it everywhere. Not worth the effort IMO.